cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaConnectConfigurationBuilderTest.java

```diff
@@ -514,4 +532,217 @@ public void withPluginPath() {
+                "group.id=connect-cluster",
+                "offset.storage.topic=connect-cluster-offsets",
+                "config.storage.topic=connect-cluster-configs",
+                "status.storage.topic=connect-cluster-status",
+                "key.converter=org.apache.kafka.connect.json.JsonConverter",
+                "value.converter=org.apache.kafka.connect.json.JsonConverter",
+                "metric.reporters=kafka.metrics.KafkaPrometheusMetricsReporter",
+                "admin.metric.reporters=kafka.metrics.KafkaPrometheusMetricsReporter",
+                "producer.metric.reporters=kafka.metrics.KafkaPrometheusMetricsReporter",
+                "consumer.metric.reporters=kafka.metrics.KafkaPrometheusMetricsReporter"));
+    }
+
+    @ParallelTest
+    public void testNullUserConfigurationWithJmxMetricsReporter() {
+        String configuration = new KafkaConnectConfigurationBuilder(BOOTSTRAP_SERVERS)
+                .withUserConfiguration(null, true, false)
+                .build();
+
+        assertThat(configuration, isEquivalent("bootstrap.servers=my-cluster-kafka-bootstrap:9092",
+                "security.protocol=PLAINTEXT",
+                "producer.security.protocol=PLAINTEXT",
+                "consumer.security.protocol=PLAINTEXT",
+                "admin.security.protocol=PLAINTEXT",
+                "config.providers=strimzienv,strimzifile,strimzidir",
+                "config.providers.strimzienv.class=org.apache.kafka.common.config.provider.EnvVarConfigProvider",
+                "config.providers.strimzienv.param.allowlist.pattern=.*",
+                "config.providers.strimzifile.class=org.apache.kafka.common.config.provider.FileConfigProvider",
+                "config.providers.strimzidir.class=org.apache.kafka.common.config.provider.DirectoryConfigProvider",
+                "config.providers.strimzidir.param.allowed.paths=/opt/kafka",
+                "group.id=connect-cluster",
+                "offset.storage.topic=connect-cluster-offsets",
+                "config.storage.topic=connect-cluster-configs",
+                "status.storage.topic=connect-cluster-status",
+                "key.converter=org.apache.kafka.connect.json.JsonConverter",
+                "value.converter=org.apache.kafka.connect.json.JsonConverter",
+                "metric.reporters=org.apache.kafka.common.metrics.JmxReporter",
+                "admin.metric.reporters=org.apache.kafka.common.metrics.JmxReporter",
+                "producer.metric.reporters=org.apache.kafka.common.metrics.JmxReporter",
+                "consumer.metric.reporters=org.apache.kafka.common.metrics.JmxReporter"));
+    }
+
+    static Stream<Arguments> userConfigurationWithMetricsReporters() {
+        Map<String, Object> configMap = new HashMap<>();
+        //configMap.put("auto.create.topics.enable", "false");
+        configMap.put("metric.reporters", "my.domain.CustomMetricReporter");
```

Comment:
Where are all the other configurations with prefixes?

Reply:
This is still not correct. We are only interested in `metric.reporters` behavior here.